src/components/course/data/hooks.jsx

```diff
@@ -493,6 +493,30 @@ export const useTrackSearchConversionClickHandler = ({ href, eventName }) => {
   return handleClick;
 };
 
+/**
+ * Returns a function to be used as a click handler that emits an optimizely license subsidy enrollment click event.
+ *
+ * @returns Click handler function for clicks on enrollment buttons when a learner possesses a license subsidy.
+ */
+export const useOptimizelyLicenseSubsidyEnrollmentClickHandler = ({ href, courseRunKey }) => {
+  const handleClick = useCallback(
+    (e) => {
+      // If tracking is on a link with an external href destination, we must intentionally delay the default click
+      // behavior to allow enough time for the async analytics event call to resolve.
+      if (href) {
+        e.preventDefault();
+        setTimeout(() => {
+          global.location.href = href;
+        }, CLICK_DELAY_MS);
+      }
+      pushEvent(EVENTS.LICENSE_SUBSIDY_ENROLLMENT_CLICK, { courseKey: courseRunKey });
+    },
+    [courseRunKey, href],
+  );
+
+  return handleClick;
```

Comment:
@iloveagent57 Even though I have merged this but I not sure if this hook is need at all? My understanding is that I just need to do `pushEvent(EVENTS.LICENSE_SUBSIDY_ENROLLMENT_CLICK, { courseKey: courseRunKey });` where this hook is being used? Are we adding delay just to ensure that analytics call gets completed? Can you help understand how this is working?

Reply:
Yeah, that's right - there needs to be a long enough (asynchronous) delay that the events can be transmitted to Optimizely (or segment, or any other system that stores events for us).  We want the calls to those systems to resolve before the user is redirected anywhere else as a result of clicking the button.

https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#working_with_asynchronous_functions

Reply:
Thank you! This is clear now. One more question. After the delay, when `global.location.href = href` gets executed by every hook, will that redirects to same url multiple times or `global.location.href` is smart enough to redirect only once?